services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/mapper/BeanTableSchemaTest.java

```diff
@@ -1160,4 +1161,22 @@ public void emptyConverterProviderList_correct_whenAttributeConvertersAreSupplie
         assertThat(reverse.getId(), is(equalTo("id-value-custom")));
         assertThat(reverse.getIntegerAttribute(), is(equalTo(133)));
     }
+
+    @Test
+    public void fluentSetterBean_correctlyMapsBeanAttributes() {
+        BeanTableSchema<FluentSetterBean> beanTableSchema =
+            BeanTableSchema.create(FluentSetterBean.class);
+
+        FluentSetterBean fluentSetterBean = new FluentSetterBean()
+            .setAttribute1(1)
+            .setAttribute2("testAttribute2")
+            .setAttribute3("testAttribute3");
+
+        Map<String, AttributeValue> itemMap = beanTableSchema.itemToMap(fluentSetterBean, false);
```

Comment:
Can we verify via roundtrip from map to item using `beanTableSchema.mapToItem()`?

```java
FluentSetterBean roundTrip = beanTableSchema.mapToItem(itemMap);
assertThat(roundTrip, isEqualTo(fluentSetterBean));
```